tests/test_distributions.py

```diff
@@ -200,11 +200,19 @@ def test_eq_ne_hash():
     d0 = distributions.UniformDistribution(low=1, high=2)
     d1 = distributions.UniformDistribution(low=1, high=3)
     assert d0 != d1
+    assert not d0 == d1
     assert hash(d0) != hash(d1)
 
-    # Different classes.
+    # Different distribution classes.
     d2 = distributions.IntUniformDistribution(low=1, high=2)
     assert d0 != d2
+    assert not d0 == d2
+
+    # Different types.
+    assert d0 != 1
+    assert not d0 == 1
+    assert d0 != 'foo'
+    assert not d0 == 'foo'
 
     # In the implementation of `__hash__`, only attributes are considered.
     assert hash(d0) == hash(d2)
```

Comment:
I'm starting to wonder whether this is "correct". It seems counterintuitive considering the fact that `d0 != d2`, but is this correct?

Reply:
I understand your concern.
You can find the past discussion about this implementation [here](https://github.com/optuna/optuna/pull/573#discussion_r344566282), and if you come up with better implementation, please create a PR.

Reply:
Thanks for the reference, this is basically the implementation that I was imaging but I'll have a proper look at it. Sorry by the way for change that made this test to fail. I was testing it locally and committed it by mistake.